Docs/Api/CSharpDriverDocs.shfbproj

```diff
@@ -106,6 +107,15 @@
     <DisableCodeBlockComponent>False</DisableCodeBlockComponent>
     <CppCommentsFixup>False</CppCommentsFixup>
     <CleanIntermediates>True</CleanIntermediates>
+    <PlugInConfigurations>
```

Comment:
Related comment: https://github.com/EWSoftware/SHFB/issues/67#issuecomment-92990671.
NOTE: If I add a `MongoDB.Libmongocrypt` just as a new `DocumentationSource` into `CSharpDriverDocs.shfbproj`, then I see a lot of warnings from the wrapper project as warnings during a build.

Reply:
I am unfamiliar with what you are doing in lines 110-118.

As an alternative is there any way to tell Sandcastle to *ignore* `MongoDB.LibMongocrypt`?

Reply:
OK. Can you explain what this is doing so we don't all have to go read the github page.

Reply:
As I understand, the warning

    Unknown reference link target &#39;T:MongoDB.Libmongocrypt.CryptClient&#39

 was related to the fact that the `CryptClient` was defined in an external library (which wasn't added via `DocumentationSource`). I'm not sure how it affected generated documents (after quick check, I didn't find any differences in docs before and after this change), but it looks like `Sandcastle` requires that all used types(their assemblies) should be mentioned in `*.shfbproj`.

In this change, I've created a different `*.shfbproj` which contains `DocumentationSource` with `MongoDB.Libmongocrypt.dll` and then just added a reference on this `*.shfbproj` via `PlugInConfigurations`. So, I consider this change just in the same way, as just adding `using namespace_name` in `.cs` files.

Reply:
So is this just a very roundabout way of suppressing a warning?

We create a whole new `.shfbproj` *just* so that we can add a reference to it from our current `.shfbproj`?

But the new `LibMongocryptDocs.shfbproj` is just a *dummy* project that we never actually build?

Are you sure there aren't a bunch of broken links pointing to this new `.shfbproj` that we never build?

I'm just guessing on what I'm saying above. I think a more detailed explanation of what is happening here is needed.

Reply:
>So is this just a very roundabout way of suppressing a warning?

yes

>We create a whole new .shfbproj just so that we can add a reference to it from our current .shfbproj?

yes

>Are you sure there aren't a bunch of broken links pointing to this new .shfbproj that we never build?

AFAIK, yes. We will see this page regardless of whether we created and use a new `.shfbproj` or no.
![image](https://user-images.githubusercontent.com/44870738/67204064-550e4580-f415-11e9-9f63-82b3836171e4.png)

Reply:
> I think a more detailed explanation of what is happening here is needed.

According to this doc page: https://ewsoftware.github.io/SHFB/html/15b6b7be-3778-4487-b524-d558d02e6460.htm and in particular this paragraph:

>The None link type can also be specified to render the references as non-clickable labels. This may be of use in reducing the number of warnings about unresolved reference links in a build but is not really needed as unresolved links are automatically rendered as non-clickable labels.

I think this plugin (`Additional Reference Links`) with the option `htmlSdkLinkType="None"` is expected to be used exactly for our target (suppressing warnings like `Fix `Unknown reference link target`). But the same time this the only benefit that we have here, because unresolved links are rendered as non-clickable labels by default (without any plugins). 

So, the question is do we want to suppress the message during building and have the new `*.shfbproj` or we can ignore this warning, but we will see it in all builds.

Reply:
Yes we want to suppress warning messages.

It's annoying that the only way to do that is to create a dummy `.shfbproj` file that otherwise serves no purpose as far as I can tell.

I'd like to hear a second opinion on this workaround, but I will go ahead and LGTM on the assumption that the other reviewers will be OK with this workaround (not because it's simple, but because apparently there is no other workaround).

Reply:
It is unfortunate that this appears to be the only workaround.  Thank you Dima and Robert for the excellent background information that made this PR that much easier to review! 

In the future it might be worth looking into switching to an [alternative to SandCastle](https://stackoverflow.com/questions/15316381/create-html-documentation-for-c-sharp-code). Apparently, Microsoft has ceased support for SandCastle but is now developing DocFx